repair/repair.cc

```diff
@@ -1015,6 +1017,15 @@ int repair_service::do_repair_start(sstring keyspace, std::unordered_map<sstring
     auto id = repair_tracker().next_repair_command();
     rlogger.info("repair[{}]: starting user-requested repair for keyspace {}, repair id {}, options {}", id.uuid, keyspace, id.id, options_map);
 
+    task_status.id = id.uuid;
+    task_status.state = task_manager::task_state::running;
+    task_status.is_abortable = true;
+    task_status.start_time = gc_clock::now();
+    task_status.keyspace = keyspace;
+    // TODO Deexie type, table
+    // TODO Deexie how to calculate progress for higher level tasks
```

Comment:
We can keep a set or vector of `foreign_ptr<task_ptr>` of subordinate tasks (possibly living on other shards
and register them when they are created.
If we also keep a `parent_task` backpointer (that would be disengaged for top-level tasks) and then we can use it to get the parent task_id rather than keeping it in the task_status.

If you have that you can traverse downwards to calculate progress and abort recursively.